### DISCUSSION TITLE
@xstate/react: Offer react context based mechanism to make a machine available in child components

### DISCUSSION THREAD

**Discussion by jinalskothari** [statelyai/xstate#3418] [Ideas] [answered]
Great project, thank you @davidkpiano 👍 

According to [official docs](https://xstate.js.org/docs/recipes/react.html#global-state-react-context), users are recommended to wire up react context themselves. Couldn't we offer some simple hooks-based utility to do this for them? Perhaps this is already handled in v5?

**Comment by davidkpiano** [comment, marked as answer]
Please see the RFC for this; it's something that we're planning: https://github.com/statelyai/rfcs/pull/8
